### PR TITLE
os: improve bytes handling

### PR DESCRIPTION
**Pull request by JelleZijlstra**

### CODE REVIEW

File: stdlib/os/__init__.pyi [hunk 1]
```diff
@@ -503,11 +503,14 @@ if sys.platform != "win32":
     def getenvb(key: bytes) -> bytes | None: ...
     @overload
     def getenvb(key: bytes, default: _T) -> bytes | _T: ...
+    def putenv(__name: StrOrBytesPath, __value: StrOrBytesPath) -> None: ...
+    def unsetenv(__name: StopAsyncIteration) -> None: ...
```
Comment (hauntsaninja) [inline stdlib/os/__init__.pyi:507]: Where are you getting StopAsyncIteration from?

Comment (JelleZijlstra) [reply stdlib/os/__init__.pyi:507]: lol, autocomplete
```suggestion
    def unsetenv(__name: StrOrBytesPath) -> None: ...
```
I'm impressed this had no mypy-primer hits.

**Review by hauntsaninja — APPROVED** [review]
Everything else in here looks good!

File: stdlib/os/__init__.pyi (hunk 1 above)
Comment (hauntsaninja) [reply stdlib/os/__init__.pyi]: tsk, symbolic autocomplete. copilot would never

**Pull request merged**